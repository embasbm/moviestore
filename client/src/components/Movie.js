import React, { Component } from 'react';

class Movie extends Component {
  render() {
    const { movie, onremoveMovie, editingMovie} = this.props;
    return (
      <div className="single-movie">
        <h4>{movie.title}</h4>
        <p>{movie.text}</p>
        <button onClick={() => onremoveMovie(movie.id)}>Erase</button>
        <button onClick={() => editingMovie(movie.id)}>Edit</button>
      </div>
    )
  }
}

export default Movie;
