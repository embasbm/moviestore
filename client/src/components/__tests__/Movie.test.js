import React from 'react';
import axios from "axios";
const ReactTestRenderer = require('react-test-renderer');
import Movie from '../Movie';

jest.mock('axios');

describe('Movie', () => {
  const movie = {
    "id": 1,
    "title": "spiderman",
    "text": "Omnis repellat harum. Error beatae placeat. Ut minima nam."
  };
  axios.get.mockImplementation(() => Promise.resolve(resp));

  it('Should compare Movie component with a snapshot', () => {
    const component = ReactTestRenderer.create(<Movie movie={movie}/>);
    const json = component.toJSON();
    expect(json).toMatchSnapshot();
  })
});
