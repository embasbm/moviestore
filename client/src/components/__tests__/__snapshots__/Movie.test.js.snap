// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Movie Should compare Movie component with a snapshot 1`] = `
<div
  className="single-movie"
>
  <h4>
    spiderman
  </h4>
  <p>
    Omnis repellat harum. Error beatae placeat. Ut minima nam.
  </p>
  <button
    onClick={[Function]}
  >
    Erase
  </button>
</div>
`;
